parachain/pallets/verifier-lightclient/src/lib.rs

```diff
@@ -119,26 +152,44 @@ decl_module! {
 impl<T: Trait> Module<T> {
 	// Validate an Ethereum header for import
 	fn validate_header_to_import(header: &EthereumHeader, proof: &[EthashProofData]) -> DispatchResult {
-		ensure!(
-			Headers::<T>::contains_key(header.parent_hash),
-			Error::<T>::MissingParentHeader,
-		);
-
 		let hash = header.compute_hash();
 		ensure!(
 			!Headers::<T>::contains_key(hash),
 			Error::<T>::DuplicateHeader,
 		);
 
+		let parent = Headers::<T>::get(header.parent_hash)
+			.ok_or(Error::<T>::MissingParentHeader)?
+			.header;
+
+		let finalized_header_id = FinalizedBlock::get();
+		ensure!(
+			header.number > finalized_header_id.number,
+			Error::<T>::AncientHeader,
+		);
+
+		// TODO: limit N where N = header.number - finalized_header.number
+		// to avoid iterating over long chains here
+		for (ancestor_hash, ancestor) in ancestry::<T>(header.parent_hash) {
+			if ancestor.number > finalized_header_id.number {
```

Comment:
This code is duplicated in `get_best_finalized_header`. Could use the `best_descendant` method I wrote in another comment to factor out this logic?